modules/loader/src/main/java/com/opengamma/strata/loader/csv/TradeCsvLoader.java

```diff
@@ -0,0 +1,422 @@
+      }
+    }
+    return ValueWithFailures.of(trades, failures);
+  }
+
+  // parse the trade info
+  private TradeInfo parseTradeInfo(CsvRow row) {
+    TradeInfoBuilder infoBuilder = TradeInfo.builder();
+    String scheme = row.findField(ID_SCHEME_FIELD).orElse("OG-Trade");
+    row.findValue(ID_FIELD).ifPresent(id -> infoBuilder.id(StandardId.of(scheme, id)));
+    row.findValue(TRADE_DATE_FIELD).ifPresent(dateStr -> infoBuilder.tradeDate(parseDate(dateStr)));
+    row.findValue(TRADE_TIME_FIELD).ifPresent(timeStr -> infoBuilder.tradeTime(LocalTime.parse(timeStr)));
+    row.findValue(TRADE_ZONE_FIELD).ifPresent(zoneStr -> infoBuilder.zone(ZoneId.of(zoneStr)));
+    return infoBuilder.build();
+  }
+
+  //-------------------------------------------------------------------------
+  // parses a date
+  static boolean parseBoolean(String str) {
+    switch (str.toUpperCase(Locale.ENGLISH)) {
+      case "TRUE":
+      case "T":
+      case "YES":
+        return true;
+      case "FALSE":
+      case "F":
+      case "NO":
+        return false;
+      default:
+        throw new IllegalArgumentException("Unknown boolean value, must 'True' or 'False' but was '" + str + "'");
+    }
+  }
+
+  // parses a double
+  static Double parseDouble(String s) {
+    return new BigDecimal(s).doubleValue();
+  }
+
+  // parses a double as a percentage
+  static Double parseDoublePercent(String s) {
+    return new BigDecimal(s).movePointLeft(2).doubleValue();
+  }
+
+  // parses a date
+  static LocalDate parseDate(String str) {
+    try {
+      // yyyy-MM-dd
+      if (str.length() == 10 && str.charAt(4) == '-' && str.charAt(7) == '-') {
+        return LocalDate.parse(str, YYYY_MM_DD_DASH);
+      }
+      // yyyy/MM/dd
+      if (str.length() == 10 && str.charAt(4) == '/' && str.charAt(7) == '/') {
+        return LocalDate.parse(str, YYYY_MM_DD_SLASH);
+      }
+      // dd/MM/yy
+      // dd/MM/yyyy
+      if (str.length() >= 6 && str.charAt(2) == '/' && str.charAt(5) == '/') {
+        if (str.length() == 8) {
+          return LocalDate.parse(str, DD_MM_YY_SLASH);
+        } else {
+          return LocalDate.parse(str, DD_MM_YYYY_SLASH);
+        }
+      }
+      // d-MMM-yyyy
+      if (str.length() >= 10 && str.charAt(str.length() - 5) == '-') {
+        return LocalDate.parse(str, D_MMM_YYYY_DASH);
+      }
+      if (str.length() == 8 && Character.isDigit(str.charAt(2))) {
+        // yyyyMMdd (and all others)
+        return LocalDate.parse(str, YYYYMMDD);
+      }
+      // dMMMyyyy (and all others)
+      return LocalDate.parse(str, D_MMM_YYYY_NODASH);
+
+    } catch (DateTimeParseException ex) {
+      throw new IllegalArgumentException(
+          "Unknown date format, must be formatted as " +
+              "yyyy-MM-dd, yyyyMMdd, dd/MM/yyyy, yyyy/MM/dd, 'd-MMM-yyyy' or 'dMMMyyyy' but was: " + str,
+          ex);
```

Comment:
Exceptions are only including the recommended inputs, not everything we accept ;-)